src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
+        auto guid = DEFAULT_WINDOWS_POWERSHELL_GUID;
 
-        // Load profiles from dynamic profile generators. _userSettings should be
-        // created by now, because we're going to check in there for any generators
-        // that should be disabled (if the user had any settings.)
-        resultPtr->_LoadDynamicProfiles();
-        try
+        for (const auto& profile : inboxSettings.profiles)
         {
-            resultPtr->_LoadFragmentExtensions();
+            if (profile->Name() == preferredPowershellProfile)
+            {
+                guid = profile->Guid();
+                break;
+            }
         }
-        CATCH_LOG();
 
-        if (!fileHasData)
+        userSettings.globals->DefaultProfile(guid);
+    }
+
+    // 2.
+    {
+        for (const auto& profile : userSettings.profiles)
         {
-            // We didn't find the user settings. We'll need to create a file
-            // to use as the user defaults.
-            // For now, just parse our user settings template as their user settings.
-            auto userSettings{ resultPtr->_ApplyFirstRunChangesToSettingsTemplate(UserSettingsJson) };
-            resultPtr->_ParseJsonString(userSettings, false);
-            needToWriteFile = true;
+            if (profile->Guid() == DEFAULT_COMMAND_PROMPT_GUID)
+            {
+                profile->Name(RS_(L"CommandPromptDisplayName"));
+                break;
+            }
         }
+    }
+}
 
-        try
+// Adds profiles from .inboxSettings as parents of matching profiles in .userSettings.
+// That way the user profiles will get appropriate defaults from the generators (like icons and such).
+// If a matching profile doesn't exist yet in .userSettings, one will be created.
+void SettingsLoader::LayerInboxOntoUser()
+{
+    for (const auto& generatedProfile : inboxSettings.profiles)
+    {
+        if (const auto [it, inserted] = userSettings.profilesByGuid.emplace(generatedProfile->Guid(), generatedProfile); !inserted)
         {
-            // See microsoft/terminal#2325: find the defaultSettings from the user's
-            // settings. Layer those settings upon all the existing profiles we have
-            // (defaults and dynamic profiles). We'll also set
-            // _userDefaultProfileSettings here. When we LayerJson below to apply the
-            // user settings, we'll make sure to use these defaultSettings _before_ any
-            // profiles the user might have.
-            resultPtr->_ApplyDefaultsFromUserSettings();
-
-            // Apply the user's settings
-            resultPtr->LayerJson(resultPtr->_userSettings);
+            it->second->InsertParent(generatedProfile);
         }
-        catch (...)
+        else
         {
-            _CatchRethrowSerializationExceptionWithLocationInfo(resultPtr->_userSettingsString);
+            userSettings.profiles.emplace_back(CreateChild(generatedProfile));

```

Comment:
nit: a comment explaining that the profile has already been added to profilesByGuid would not go unmissed :)